src/env-inl.h

```diff
@@ -325,6 +326,34 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline void Environment::print_sync_warn() const {
+  if (!warn_on_sync_)
+    return;
+
+  v8::HandleScope handle_scope(isolate_);
+  v8::Local<v8::StackTrace> stack = v8::StackTrace::CurrentStackTrace(
+      isolate_, 10, v8::StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
```

Comment:
That would make things much easier. Though there's a slight style issue that doesn't sit well with me. Here's an example output:
```
WARNING: Detected use of sync API           
pbkdf2 (crypto.js:592:20)                   
Object.exports.pbkdf2Sync (crypto.js:583:10)
null._onTimeout (/tmp/x.js:8:10)            
Timer.listOnTimeout (timers.js:89:15)       
```
I would prefer the stack trace to look more like an error's stack trace with `    at` at the beginning of each line. But reimplementing the entire formatting would be a pain. Thoughts?